From a675ba69fdcb54d8a5e2f3a5e6813002841788ef Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 21 Oct 2011 23:34:07 +0800
Subject: [PATCH 163/280] ENGR00160547 vpu: Change SWReset to do whole vpu
 hardware reset on i.mx6q

Need to do vpu hardware reset in vpu_SWReset function implementation, then
vpu can go to correct status when vpu is blocked.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 9783e8f..712c915 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -249,16 +249,22 @@ RetCode vpu_SWReset(DecHandle handle, int index)
 
 	if (*ppendingInst && (pCodecInst != *ppendingInst))
 		return RETCODE_FAILURE;
-	else {
+	else if (*ppendingInst) {
 		/* Need to unlock VPU since mutex is locked when StartOneFrame */
 		UnlockVpu(vpu_semap);
+		*ppendingInst = 0;
 	}
 
 	if (!LockVpu(vpu_semap))
 		return RETCODE_FAILURE_TIMEOUT;
 
 	if (cpu_is_mx6q()) {
-		vpu_mx6q_swreset(0);
+		IOSysSWReset();
+
+		VpuWriteReg(BIT_BUSY_FLAG, 1);
+		VpuWriteReg(BIT_CODE_RUN, 1);
+		while (vpu_IsBusy());
+
 		UnlockVpu(vpu_semap);
 		return RETCODE_SUCCESS;
 	}
-- 
1.8.0

